pyccl/halos/pk_4pt.py

```diff
@@ -505,3 +585,976 @@ def _logged_output(*arrs, log):
+            second moment will be used, corresponding to the
+            products of the means of both profiles.
+        prof13_2pt (:class:`~pyccl.halos.profiles_2pt.Profile2pt`):
+            same as `prof12_2pt` for `prof` and `prof3`.
+        prof14_2pt (:class:`~pyccl.halos.profiles_2pt.Profile2pt`):
+            same as `prof14_2pt` for `prof` and `prof4`.
+        prof24_2pt (:class:`~pyccl.halos.profiles_2pt.Profile2pt`):
+            same as `prof14_2pt` for `prof2` and `prof4`.
+        prof32_2pt (:class:`~pyccl.halos.profiles_2pt.Profile2pt`):
+            same as `prof14_2pt` for `prof3` and `prof2`.
+        prof34_2pt (:class:`~pyccl.halos.profiles_2pt.Profile2pt`):
+            same as `prof34_2pt` for `prof3` and `prof4`.
+        p13_of_k_a (:class:`~pyccl.pk2d.Pk2D`): same as p12_of_k_a for 13
+        p14_of_k_a (:class:`~pyccl.pk2d.Pk2D`): same as p12_of_k_a for 14
+        p_of_k_a (:class:`~pyccl.pk2d.Pk2D`): a `Pk2D` object to
+            be used as the linear matter power spectrum. If `None`, the power
+            spectrum stored within `cosmo` will be used.
+        a_arr (array): an array holding values of the scale factor
+            at which the trispectrum should be calculated for
+            interpolation. If `None`, the internal values used
+            by `cosmo` will be used.
+        lk_arr (array): an array holding values of the natural
+            logarithm of the wavenumber (in units of Mpc^-1) at
+            which the trispectrum should be calculated for
+            interpolation. If `None`, the internal values used
+            by `cosmo` will be used.
+        extrap_order_lok (int): extrapolation order to be used on
+            k-values below the minimum of the splines. See
+            :class:`~pyccl.tk3d.Tk3D`.
+        extrap_order_hik (int): extrapolation order to be used on
+            k-values above the maximum of the splines. See
+            :class:`~pyccl.tk3d.Tk3D`.
+        use_log (bool): if `True`, the trispectrum will be
+            interpolated in log-space (unless negative or
+            zero values are found).
+
+    Returns:
+        :class:`~pyccl.tk3d.Tk3D`: 2-halo trispectrum.
+    """
+    if lk_arr is None:
+        lk_arr = cosmo.get_pk_spline_lk()
+    if a_arr is None:
+        a_arr = cosmo.get_pk_spline_a()
+
+    tkk_2h_22 = halomod_trispectrum_2h_22(cosmo, hmc, np.exp(lk_arr), a_arr,
+                                          prof, prof2=prof2,
+                                          prof3=prof3, prof4=prof4,
+                                          prof13_2pt=prof13_2pt,
+                                          prof14_2pt=prof14_2pt,
+                                          prof24_2pt=prof24_2pt,
+                                          prof32_2pt=prof32_2pt,
+                                          p_of_k_a=p_of_k_a)
+
+    tkk_2h_13 = halomod_trispectrum_2h_13(cosmo, hmc, np.exp(lk_arr), a_arr,
```

Comment:
I would suggest that, if the `_22` and `_13` components of this are not meant to be called separately by users, we just make them private methods.

Reply:
Done